scripts/generator/graphql_generator/csharp/ShopifyBuy.cs.erb

```diff
@@ -849,9 +856,17 @@ namespace Shopify.Unity {
                 (response) => {
                     var error = (ShopifyError)response;
                     if (error != null) {
-                        callback(null, error);
+                        callback(null, error, null);
                     } else {
-                        callback((List<Product>) response.data.shop().products(), null);
+                        string lastCursor = null;
+
+                        if (response.data.shop().products().pageInfo().hasNextPage()) {
+                            List<ProductEdge> productEdges = response.data.shop().products().edges();
```

Comment:
Is there a chance that productEdges has no elements? Not sure since we do the `hasNextPage` check here. If so we'll crash below.

Reply:
Yeh I guess the only way you could break this is by doing something like `first: 0`